src/helpers/workspace.ts

```diff
@@ -185,3 +189,120 @@ async function selectTruffleConfigFromQuickPick(workspaces: TruffleWorkspace[]):
 
   return result.truffleWorkspace;
 }
+
+export namespace AbstractWorkspaceManager {
```

Comment:
It is abstract. 

It is provider agnostic hence the name. There are multiple resolvers inside it so in theory it would remove all the other code above in time and make the resolving of config and uri locations agnostic. so if we find a truffle config in folder A, then folder A is a truffle based SDK command driven workspace/folder. Folder B could hh or something else?

Does that make sense?

Namespace was here to just encapsulate it in this file v's making another workspace file.